packages/esm-patient-immunizations-app/src/hooks/useImmunizations.ts

```diff
@@ -1,16 +1,13 @@
-import { fhirBaseUrl, openmrsFetch } from '@openmrs/esm-framework';
-import useSWR from 'swr';
-import { type FHIRImmunizationBundle } from '../types/fhir-immunization-domain';
+import { fhirBaseUrl, openmrsFetch, useFhirFetchAll } from '@openmrs/esm-framework';
+import { type FHIRImmunizationResource } from '../types/fhir-immunization-domain';
 import { mapFromFHIRImmunizationBundle } from '../immunizations/immunization-mapper';
 
 export function useImmunizations(patientUuid: string) {
   const immunizationsUrl = `${fhirBaseUrl}/Immunization?patient=${patientUuid}`;
 
-  const { data, error, isLoading, isValidating, mutate } = useSWR<{ data: FHIRImmunizationBundle }, Error>(
-    immunizationsUrl,
-    openmrsFetch,
-  );
-  const existingImmunizations = data ? mapFromFHIRImmunizationBundle(data.data) : null;
+  const { data, error, isLoading, isValidating, mutate } = useFhirFetchAll(immunizationsUrl);
+
+  const existingImmunizations = data ? mapFromFHIRImmunizationBundle(data as FHIRImmunizationResource[]) : [];
```

Comment:
thanks @Muppasanipraneeth for the PR,, i see this blindly casts data as `FHIRImmunizationResource[]` without verifying the actual structure, we can do some explicit structure validation that checks for both array and bundle structures ie 

```
let existingImmunizations: ImmunizationGrouped[] = [];
try {
  if (data) {
    const resources = Array.isArray(data)
      ? data 
      : data?.entry?.map(entry => entry.resource) ?? []; // Extract from Bundle

    existingImmunizations = mapFromFHIRImmunizationBundle(resources);
  }
} catch (e) {
  console.error('Immunization data processing error:', e);
}

```